healthcare-management/src/app/components/signup/signup.component.ts

```diff
@@ -0,0 +1,75 @@
+import { Component } from '@angular/core';
+import { MatCard } from '@angular/material/card';
+import { MatFormField, MatLabel } from '@angular/material/form-field';
+import { MatButton } from '@angular/material/button';
+import { MatInput } from '@angular/material/input';
+import { MatSelect, MatOption } from '@angular/material/select';
+import { FormsModule, ReactiveFormsModule } from '@angular/forms';
+import { CommonModule } from '@angular/common';
+import { AuthService} from '../../services/users/auth.service';
+import {Router} from '@angular/router';
+
+@Component({
+  selector: 'app-signup',
+  imports: [
+    MatCard,
+    MatFormField,
+    MatButton,
+    MatInput,
+    MatLabel,
+    MatSelect,
+    MatOption,
+    FormsModule,
+    ReactiveFormsModule,
+    CommonModule
+  ],
+  templateUrl: './signup.component.html',
+  styleUrls: ['./signup.component.scss']
+})
+export class SignupComponent {
+  email: string = '';
+  password: string = '';
+  confirmPassword: string = '';
+  firstName: string = '';
+  lastName: string = '';
+  phoneNumber: string = '';
+  dateOfBirth: string = '';
+  role: string = '';
+
+  constructor(
+    readonly authService: AuthService,
+    readonly router: Router
+  ) {
+  }
+
+  onSubmit(): void {
+    const requestData = {
+      email: this.email,
+      password: this.password,
+      firstName: this.firstName,
+      lastName: this.lastName,

```

Comment:
Aici ar fi mers sa se fi folosit un formBuilder, pentru a avea validatori in caz de nr minim de caractere, validare la email